src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -311,13 +439,16 @@ out ImmutableArray<(BoundExpression, BoundDagTemp)> bindings)
                         var evaluation = new BoundDagFieldEvaluation(syntax, field, input); // fetch the ItemN field
                         decisions.Add(evaluation);
                         var output = new BoundDagTemp(syntax, field.Type, evaluation, 0);
-                        MakeDecisionsAndBindings(output, pattern, decisions, bindings, ref discardedUseSiteDiagnostics);
+                        MakeDecisionsAndBindings(output, pattern, decisions, bindings);
                     }
                 }
+                // PROTOTYPE(patterns2): Handle the ITuple case.
                 else
                 {
-                    // PROTOTYPE(patterns2): This should not occur except in error cases. Perhaps this will be used to handle the ITuple case.
+                    // This occurs in error cases.
                     Debug.Assert(recursive.HasAnyErrors);
+                    // To prevent this pattern from subsuming other patterns and triggering a cascaded diagnostic, we add a test that will fail.
+                    decisions.Add(new BoundTypeDecision(recursive.Syntax, ErrorType(), input, hasErrors: true));
```

Comment:
>new BoundTypeDecision [](start = 34, length = 21)

Is this additional decision observable, from the semantic model in particular?

Reply:
Decisions are not observable from the semantic model. They are an internal artifact of the compilation strategy.